collections/collections.go

```diff
@@ -0,0 +1,90 @@
+package collections
+
+import (
+	"errors"
+	"math"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+var (
+	// ErrNotFound is returned when the provided key is not present in the StorageProvider.
+	ErrNotFound = errors.New("collections: not found")
+	// ErrEncoding is returned when something fails during key or value encoding/decoding.
+	ErrEncoding = errors.New("collections: encoding error")
+)
+
+// StorageProvider represents sdk.Context
+// it is used to avoid to reduce dependencies.
+type StorageProvider interface {
+	// KVStore returns a KVStore given its StoreKey.
+	KVStore(key storetypes.StoreKey) storetypes.KVStore
+}
+
+// Prefix defines a segregation namespace
+// for specific collections objects.
+type Prefix struct {
+	raw []byte // TODO(testinginprod): maybe add a humanized prefix?
```

Comment:
By this TODO, do you mean a string version? If so, let's make this an interface?

Reply:
I will leave it as is for now, this will be coupled with schema reflection which @aaronc is designing rn. It's just meant to make humans understand that prefix `[]byte{0x0}` is actually `Params`

Reply:
resolving, feel free to re-comment if answer is not satisfactory